src/Magnum/DartIntegration/Object.cpp

```diff
@@ -0,0 +1,209 @@
+#endif
+
+Object::Object(SceneGraph::AbstractBasicObject3D<Float>& object, SceneGraph::AbstractBasicTranslationRotation3D<Float>& transformation, dart::dynamics::ShapeNode* node, dart::dynamics::BodyNode* body): SceneGraph::AbstractBasicFeature3D<Float>{object}, _transformation(transformation), _node{node}, _body{body}, _updated(false), _updatedMesh(false) {}
+
+Object& Object::update(Trade::AbstractImporter* importer) {
+    /* If the object is has a shape and could not extract the DrawData,
+     * do not update it; i.e., the user can choose to delete it
+     */
+    if(_node && !extractDrawData(importer))
+        return *this;
+
+    /* Get transform from DART */
+    const Eigen::Isometry3d* trans;
+    if(!_node)
+        trans = &_body->getRelativeTransform();
+    else
+        trans = &_node->getRelativeTransform();
+
+    Eigen::Quaterniond quat(trans->linear());
+    Eigen::Vector3d axis(quat.x(), quat.y(), quat.z());
+    double angle = 2.*std::acos(quat.w());
+    if(std::abs(angle)>1e-5) {
+        axis = axis.array() / std::sqrt(1-quat.w()*quat.w());
+        axis.normalize();
+    }
+    else
+        axis(0) = 1.;
+
+    Eigen::Vector3d T = trans->translation();
+
+    /* Convert it to axis-angle representation */
+    Math::Vector3<Float> t(T[0], T[1], T[2]);
+    Math::Vector3<Float> u(axis(0), axis(1), axis(2));
+    Rad theta(angle);
+
+    /* Pass it to Magnum */
+    _transformation.resetTransformation()
+        .rotate(theta, u)
+        .translate(t);
+
+    /* Set update flag */
+    _updated = true;
+
+    return *this;
+}
+
+bool Object::extractDrawData(Trade::AbstractImporter* importer) {
+    _updatedMesh = false;
+
+    /* This is not a valid object */
+    if(!_node && !_body)
+        return false;
+    /* This object has no shape */
+    if(!_node)
+        return true;
+
+    unsigned int dataVariance = _node->getShape()->getDataVariance();
+
+    if(_drawData && dataVariance == dart::dynamics::Shape::DataVariance::STATIC)
+        return true;
+
+    bool firstTime = !_drawData;
+
+    dart::dynamics::ShapeNode& shapeNode = *this->shapeNode();
+    dart::dynamics::ShapePtr shape = shapeNode.getShape();
+
+    bool getMaterial = shape->checkDataVariance(dart::dynamics::Shape::DataVariance::DYNAMIC_COLOR);
+    bool getPrimitive = shape->checkDataVariance(dart::dynamics::Shape::DataVariance::DYNAMIC_PRIMITIVE);
+    bool getMesh = shape->checkDataVariance(dart::dynamics::Shape::DataVariance::DYNAMIC_VERTICES)
+                    || shape->checkDataVariance(dart::dynamics::Shape::DataVariance::DYNAMIC_ELEMENTS)
+                    || shape->checkDataVariance(dart::dynamics::Shape::DataVariance::DYNAMIC);
```

Comment:
Almost there: these three bools could be moved directly inside the `if(get...)` conditions below, as they are not used anywhere else